client/galaxy/scripts/mvc/workflow/workflow.js

```diff
@@ -1,94 +1,239 @@
+                    newName += ' shared by user ' + currentOwner;
+                }
+                wfJson.name = newName;
+                self.collection.create(wfJson, { at: 0,
+                                                  wait: true,
+                                                  success: function() {
+                                                      mod_toastr.success("Successfully copied workflow '" + oldName + "' to '" + newName + "'")
+                                                  },
+                                                  error : function(err) {
+                                                      console.log('error callback');
+                                                      // this error message for dev only
+                                                      alert('There was an error. See console for details');
+                                                      console.log(err);
+                                                  }
+                });
+            }).error(function(jqXHR, textStatus, errorThrown) {
+                        mod_toastr.error(jqXHR.responseJSON.err_msg);
+            })
+        },
+
+        _rowTemplate: function() {
+            var show = this.model.get("show_in_tool_panel");
+            var wfId = this.model.id;
+            var checkboxHtml = '<input id="show-in-tool-panel" type="checkbox" class="show-in-tool-panel" '+ ( show ? 'checked="' + show + '"' : "" ) +' value="' + wfId + '">';
+            var trHtml =  '<td>' +
+                             '<div class="dropdown">' +
+                                 '<button class="menubutton" type="button" data-toggle="dropdown">' +
+                                     _.escape( this.model.get("name") ) + '<span class="caret"></span>' +
+                                 '</button>' +
+                                 this._templateActions( ) +
+                             '</div>' +
+                          '</td>' +
+                          '<td><span>' + '<div class="' + wfId + ' tags-display"></div>' + '</td>' +
+                          '<td>' + ( this.model.get('owner') === Galaxy.user.attributes.username ? "You" : this.model.get('owner') ) +'</span></td>' +
+                          '<td>' + this.model.get("number_of_steps") + '</td>' +
+                          '<td>' + ( this.model.get("published") ? "Yes" : "No" ) + '</td>' +
+                          '<td>'+ checkboxHtml + '</td>';
+            return trHtml;
+        },
+
+        renderTagEditor: function(){
+            var TagEditor = new TAGS.TagsEditor({
+            model           : this.model,
+            el              : $.find( '.' + this.model.id + '.tags-display' ),
+            workflow_mode   : true });
+            TagEditor.toggle( true );
+            TagEditor.render();
+        },
+
+        /** Template for user actions for workflows */
+        _templateActions: function( ) {
```

Comment:
I think using the underscore templating would be helpful here but not necessary.

Reply:
Can you point me to an example in the codebase ?

Reply:
https://github.com/galaxyproject/galaxy/blob/dev/client/galaxy/scripts/mvc/history/history-view-edit.js#L562

https://github.com/galaxyproject/galaxy/blob/dev/client/galaxy/scripts/mvc/library/library-librarytoolbar-view.js#L182

(I did not even know Carl wrote a helper for that template wrapping)